### PR TITLE
support creating add-ons on the target iso (bsc #991935)

### PR DESCRIPTION
**Pull request by wfeldt**
As a result of bsc#991935 automatically adding add-ons via
add_on_products.xml during installation works again. This patch adds support
for this feature to mksusecd.

### CODE REVIEW

File: mksusecd [hunk 1]
```diff
@@ -3124,3 +3190,128 @@ sub replace_kernel_mods
...
+            <name>$name</name>
+            <url>relurl://$dir?alias=$alias</url>
+            <priority config:type="integer">$prio</priority>
+            <ask_user config:type="boolean">false</ask_user>
+            <selected config:type="boolean">true</selected>
+            <check_name config:type="boolean">false</check_name>
+        </product_item>
+# product
+
+  # inhect the new product at the end of the list
```
Comment (imobachgs) [inline mksusecd]: typo: inject

File: mksusecd [hunk 2]
```diff
@@ -538,6 +554,17 @@ Create ISO image:
...
+                                the main product.
+                                Note: this option takes a variable number of arguments. So
+                                it may be necessary to terminate the arg list with '--'.
+      --addon-name NAME         Use NAME as the add-on name.
+                                If unset, the auto-generated name 'Add-On NUM' is used, with NUM
+                                set to the smallest number that avoids name conflicts.
+      --addon-alias ALIAS       Set repo alias to ALIAS.
+                                If unset, an alias based on the repo name is generated.
+      --addon-prio NUM          Set add-on repository priority to NUM; lower NUM means higher
+                                priority (default: 60).
```
Comment (lslezak) [inline mksusecd]: Maybe this is too long but I'd add something like this to the description:

> The default installation repositories have priority 99. Any smaller number for the add-on repo will prefer the add-on packages even though the package version number is smaller than in the standard repository.

Or put it into a documentation...

File: mksusecd [hunk 3]
```diff
@@ -1316,6 +1354,34 @@ sub run_isozipl
...
+# run_createrepo(repo_dir)
+#
+# Run 'createrepo' on repo_dir to create a repo-md repo.
+#
+sub run_createrepo
+{
+  my $dir = $_[0];
+  my $ok;
+
+  my $cmd = "createrepo --simple-md-filenames $dir";
```
Comment (lslezak) [inline mksusecd]: NP: Um, what if `$dir` contains whitespace?

Comment (wfeldt) [inline mksusecd]: currently, it can't (is a generated name that doesn't have white space)

**Comment by imobachgs** [comment]
Apart from a minor typo in a comment, LGTM. But let's wait for @lslezak.

File: mksusecd [hunk 4]
```diff
@@ -593,6 +621,25 @@ Signing notes:
     If there's no 'sign-key' option, a transient key is created. The public
     part is added to the initrd and the key is deleted.
 
+Add-on notes:
```
Comment (lslezak) [inline mksusecd]: Nice! :+1:

**Comment by lslezak** [comment]
LGTM

**Pull request merged**